src/directwrite/mod.rs

```diff
@@ -172,11 +166,11 @@ impl crate::Rasterize for DirectWriteRasterizer {
         let line_height = f64::from(ascent - descent + line_gap);
 
         // Since all monospace characters have the same width, we use `!` for horizontal metrics.
-        let c = '!';
-        let glyph_index = self.get_glyph_index(face, c)?;
+        let character = '!';
+        let glyph_index = self.get_glyph_index(face, character).unwrap_or(0);
 
         let glyph_metrics = face.get_design_glyph_metrics(&[glyph_index], false);
-        let hmetrics = glyph_metrics.first().ok_or_else(|| Error::MissingGlyph(c))?;
+        let hmetrics = glyph_metrics.first().ok_or(Error::MissingSizeMetrics)?;
```

Comment:
I'm not sure if `MissingSizeMetrics` is the right call here, does this error only happen when the glyph index doesn't exist or is there another case?

Windows docs don't say anything, so I'm fine with "Don't know, so I'm just being cautious with a dedicated error type".

Reply:
Well, I've assumed that if you can't get metrics it's a metrics issue.

Reply:
Well `Missing` indicates a specific metrics issue though.

Reply:
Yeah yeah, we try to get metrics one line above, then check if we actually got, if not it's `Missing`, seems fine to me?

Reply:
The error isn't that it's missing though, the error is that we couldn't retrieve it. Why we weren't able to do so we don't know.

Missing metrics would indicate that the font doesn't have any metrics to be retrieved, but we don't know that.

Reply:
With no other platform emitting this error, I feel like the best option is just to move this to the platform specific errors. There's no good way to handle this and it's just more likely to confuse anyone than to actually help.

I don't think this is something we've ever run into, so it's not unlikely that you'll only get to this case with weird setups like fonts that don't have the character specified or just not specifying any character at all.

Reply:
>With no other platform emitting this error, I feel like the best option is just to move this to the platform specific errors. There's no good way to handle this and it's just more likely to confuse anyone than to actually help.

This is not true, we emit them from FreeType backend as well, platform reports them as None to us though....